### PR TITLE
Robust attenuated sinusoid fitting

### PR DESCRIPTION
**Pull request by pathfinder49**
This function does a similar job to `rabi_flop`, thought should be more robust. I've left `fitting.rabi_flop` for now to maintain backwads compatibility.

Feedback welcome.

### CODE REVIEW

**Comment by cjbe** [comment]
@pathfinder49 in the future could you make sure you rebase your commits to make the history tidier (i.e. it won't ever be useful to go back to your 'flake8' commit, or the 'in working state, work in progress' commit).

**Comment by dnadlinger** [comment]
@cjbe: Thankfully, GitHub now has a "Squash and merge" button.

@pathfinder49: Apparently I lied, seems like test builds don't run anymore. Something broke; I'll have a look…

**Comment by dnadlinger** [comment]
Documentation is at https://oxfordiontrapgroup.github.io/oitg/, by the wa.

**Comment by dnadlinger** [comment]
Another comment trying to nudge Circle CI.

**Comment by dnadlinger** [comment]
Oh, what is mostly a copy of this is already in master.

File: oitg/fitting/attenuated_sinusoid.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    else:
+        p_dict['phi'] = np.arcsin(y0_norm)
+    return p_dict  # input is changed due to mutability of dict!
+
+
+def derived_params(p_dict, p_error_dict):
+    """calculate commonly used derived parameters and their error
+
+    This method neglects parameter covariances!"""
+    p_dict['t_pi'] = p_dict['t_dead'] + np.pi / p_dict['omega']
```
Comment (dnadlinger) [inline oitg/fitting/attenuated_sinusoid.py]: NB: This isn't actually the point of maximal population transfer for finite decay times, see rabi_flop. (There is probably an analytical solution, though.)

File: oitg/fitting/attenuated_sinusoid.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    p_dict['t_dead'] = 0.0
+    p_dict['rate'] = 0.0
+
+    min_step = np.min(t[1:] - t[:-1])
+    duration = t[-1] - t[0]
+    f_max = 0.5 / min_step
+    f_min = 0.5 / duration
+
+    omega_list = 2 * np.pi * np.linspace(f_min, f_max, int(f_max / f_min))
```
Comment (dnadlinger) [inline oitg/fitting/attenuated_sinusoid.py]: Some comments would be nice. You could have copied those over from `rabi_flop`, where I am [also suggesting the frequency range heuristic you use](https://github.com/OxfordIonTrapGroup/oitg/blob/master/oitg/fitting/rabi_flop.py#L21-L26).

File: oitg/fitting/attenuated_sinusoid.py (hunk 1 above)
Comment (pathfinder49) [reply oitg/fitting/attenuated_sinusoid.py]: I've updated it to the analytical result.

File: oitg/fitting/attenuated_sinusoid.py (hunk 2 above)
Comment (pathfinder49) [reply oitg/fitting/attenuated_sinusoid.py]: I've improved the comments and documentation. Hope it's clear enough now.

**Comment by dnadlinger** [comment]
One last thing (since naming is known to be the hardest problem in computer science): What do you think about `decaying_sinusoid`? Shorter, and arguably more descriptive (it's not just attenuated, as in a range not from -1 to 1, but the attenuation is increasing).

**Pull request merged**